Realm/RLMNetworkTransporting.h

```diff
@@ -0,0 +1,97 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+#import "RLMSyncUtil_Private.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface RLMNetworkRequestOptions : NSObject
+@property (nonatomic, copy, nullable) NSString *authorizationHeaderName;
+@property (nonatomic, copy, nullable) NSDictionary<NSString *, NSString *> *customHeaders;
+@property (nullable, nonatomic, copy) NSDictionary<NSString *, NSURL *> *pinnedCertificatePaths;
+@end
+
+typedef enum RLMHTTPMethod {
+    GET, POST, PUT, PATCH, DELETE
+} RLMHTTPMethod;
+
+/// An HTTP request that can be made to an arbitrary server.
+@interface RLMRequest : NSObject
+
+/// The HTTP method of this request.
+@property (nonatomic, assign) RLMHTTPMethod method;
+
+/// The URL to which this request will be made.
+@property (nonatomic, strong) NSString* url;
```

Comment:
Space before * rather than after in obj-c code (this is different from the pure c++ repos because we were trying to look like "normal" obj-c code and not obj-c code written by c++ people).